### PR TITLE
105 - Adjusting the code of the icon states and adding our new icons to the game.

### PR DESCRIPTION
**Pull request by PaulRitter**
ps: i added a shaft crafting system in which you can break down a shovel and attach nails & a weapon head to it
also you should probably add some more vars to the meele weapons

missing onmob sprites(clothing.dmi):
- wu-boots
- wu-boots_wrapped
- wu-boots_armored
- wu-boots_high
- wu-gasmask
- wu-armor
- wu-armor_pauldrons
- wu-helm
- wu-helm_armored
- wu-gloves
- wu-gloves_fingerless
- wu-gloves_armored
- wu-suit

missing obj sprites (objects.dmi):
- wu-boots_high
- wu-helm_armored
- wu-gloves_armored
- eb-gloves_armored

adjustments:
- clothing.dmi -> EB-helmet => EB-helm
- clothing.dmi -> EB-helmet_armored => EB-helm_armored
- clothing.dmi -> WU-armor=> EB-armor
- clothing.dmi -> WU-armor_pauldrons=> EB-armor_pauldrons
- furniture.dmi -> stained-belt => stained-bed

files finished:
- clothing.dmi
- fullscreen.dmi
- furniture.dmi

things without objs: (of files i already finished)
- clothing.dmi
  - googles_eyes
  - sterile-gloves
  - WU-cross-silver
  - WU-cross-gold
- furniture.dmi
  - cauldron
  - cauldron_water
  - cauldron_boiling
  - cauldron_stew
  - ammo-crate-5
  - ammo-crate-4
  - ammo-crate-3
  - ammo-crate-2
  - ammo-crate-1
  - chest
  - chest_open
  - alt-chest
  - alt-chest_open
  - case
  - case_open
  - locker
  - locker_open
  - alt-closet
  - alt_closet_open
  - cabinet (missing open state)
  - alt-cabinet (missing open state)
  - table *1 *2
  - alt-table *1 *2
  - metal-table *1 *2
  - furnace_on
  - furnace_off
  - fancy-table
  - alt-fancy-table
  - chair *2
  - alt-chair *2
  - bed *2
  - bed-frame *2
  - alt-bed *2
  - pillowbed *2
  - alt-pillowbed *2
  - stained-bed *2
  - stained-pillowbed *2
  - tv_off
  - tv_on
*1  (cannot use for current one cause current ones connect w/ neighbours)
*2  (cannot use for current one cause they use materials to determine icon_state)

files to do:
- general_leftinhands.dmi
- general_rightinhands.dmi
- mech_infantry.dmi
- munitions_leftinhands.dmi
- munitions_rightinhands.dmi
- munitionsx32.dmi
- munitionsx35.dmi
- objects.dmi
- storage_slots.dmi
- x64structures.dmi
- melee.dmi
- tables.dmi

types that got their icon changed, so any childtypes might have invalid icon_states
- obj/item/target
- obj/structure/closet (!!!)
- obj/structure/closet/secure_closet/medical1
- obj/structure/closet/crate
- obj/structure/reagent_dispensers/watertank
- obj/structure/reagent_dispensers/fueltank

files not listed in pic:
- storage_slots_addon.dmi

dmi files to do that i skipped cause bimmer told me to:
- barbed_wire.dmi
- cards.dmi

notes:
- do something with boots_wrapped (???)
- do something with boots_high (???)
- what the fuck was that special armor and does it need to be brought back?
- this pr changes alot of icon vars for parent obj, so some obj might become invisible through this
- also i am using placeholder icon_states, which will make even more objs invisible

files i coded/added(all in the code/FoF/* ):
- meele.dm
- nails.dm
- shaft.dm
- code/modules/clothing/suits/armor.dm

reference pic: (fullscreen is falsely marked old)
image

### CODE REVIEW

**Comment by TheBimmer231** [comment]
missing OBJs and mob sprites will be be done by the 13th of this month.
On-mob:

- [ ] wu-boots
- [ ] wu-boots_wrapped
- [ ] wu-boots_armored
- [ ] wu-boots_high
- [ ] wu-gasmask
- [ ] wu-armor
- [ ] wu-armor_pauldrons
- [ ] wu-helm
- [ ] wu-helm_armored
- [ ] wu-gloves
- [ ] wu-gloves_fingerless
- [ ] wu-gloves_armored
- [ ] wu-suit

OBJ:

- [ ] wu-boots_high
- [ ] wu-helm_armored
- [ ] wu-gloves_armored
- [ ] eb-gloves_armored

**Comment by Quantomicus** [comment]
Please provide descriptive PR names so one doesn't need to check the job listings to know what is being done.

File: icons/FoF/needs_resprite.dmi [hunk 1]
```diff
@@ -0,0 +1 @@
+this is a dumb bypass to some things not accepting the NEEDS RESPRITE define
```
Comment (Quantomicus) [inline icons/FoF/needs_resprite.dmi]: You've tested this and it creates no issues that this dmi is basically a text file?

**Comment by Quantomicus** [comment]
Also, conflicts.

File: icons/FoF/needs_resprite.dmi (hunk 1 above)
Comment (PaulRitter) [reply icons/FoF/needs_resprite.dmi]: nope

**Comment by PaulRitter** [comment]
conflicts cause some cough cought brainlet merged #26 before this pr (which wasn't supposed to happen)

**Comment by PaulRitter** [comment]
so if you ask me revert the merge, since it was unfinished anyways, and merge this

**Pull request merged**